gobblin-metastore/src/main/java/gobblin/metastore/DataSourceProvider.java

```diff
@@ -12,42 +12,21 @@
 
 package gobblin.metastore;
 
-import javax.sql.DataSource;
 import java.util.Properties;
 
-import org.apache.commons.dbcp.BasicDataSource;
-
 import com.google.inject.Inject;
-import com.google.inject.Provider;
 import com.google.inject.name.Named;
 
-import gobblin.configuration.ConfigurationKeys;
-
 
 /**
- * A provider class for {@link javax.sql.DataSource}s.
- *
- * @author ynli
+ * {@inheritDoc}
+ * @deprecated This class is replaced by {@link JobHistoryDataSourceProvider}
  */
-public class DataSourceProvider implements Provider<DataSource> {
-
-  private final BasicDataSource basicDataSource;
+@Deprecated
+public class DataSourceProvider extends JobHistoryDataSourceProvider {
```

Comment:
Actually you don't need to keep this and deprecate as this class is not part of the public API.